src/nanoarrow/common/inline_buffer.h

```diff
@@ -291,6 +291,10 @@ static inline ArrowErrorCode ArrowBufferAppendFill(struct ArrowBuffer* buffer,
                                                    uint8_t value, int64_t size_bytes) {
   NANOARROW_RETURN_NOT_OK(ArrowBufferReserve(buffer, size_bytes));
 
+  if (size_bytes == 0) {
+    return NANOARROW_OK;
+  }
+
   memset(buffer->data + buffer->size_bytes, value, size_bytes);
```

Comment:
The UB was triggered by the fact that we were making a call of `memset(NULL, ..., 0)` - the fact that `size_bytes` was 0 I don't think prevents UB from occurring when the `dst` is NULL

Reply:
The test suite is what hits this in the first place. Are you asking to split that off into its own dedicated test?

Reply:
Ah thanks - makes a lot of sense